### ISSUE TITLE
Not ready for projects with ARC enabled

### ISSUE THREAD

**Issue by renatorozas** [bjornsallarp/BSForwardGeocoder#13]
I also added the "-fno-objc-arc" to the files with the issue and commented every "release", "autorelease" and "super dealloc" line of code but I am still getting this error.

Cast of Objective-C pointer type 'NSString *' to C pointer type 'CFStringRef' (aka 'const struct __CFString *') requires a bridged cast

Cast of C pointer type 'CFStringRef' (aka 'const struct __CFString *') to Objective-C pointer type 'NSString *' requires a bridged cast

On BSFowardGeocoder.m:

NSString _encodedString = (NSString *)CFURLCreateStringByAddingPercentEscapes(NULL,
                                                                                  (CFStringRef)string,
                                                                                  NULL,
                                                                                  (CFStringRef)@"!_'();:@&=+$,/?%#[]",
                                                                                  kCFStringEncodingUTF8);

Any help?

**Comment by renatorozas** [comment]
A bridge cast is needed but I really don't know how to implement it or what it is.

http://stackoverflow.com/questions/7036350/arc-and-bridged-cast

**Comment by renatorozas** [comment]
I solved the problem by changing that piece of code for this:

NSString _encodedString = (__bridge NSString *)CFURLCreateStringByAddingPercentEscapes(NULL,
                                                                                  (__bridge CFStringRef)string,
                                                                                  NULL,
                                                                                  (CFStringRef)@"!_'();:@&=+$,/?%#[]",
                                                                                  kCFStringEncodingUTF8);

Basically changing (NSString *) for (__bridge NSString *) and (CFStringRef) for  (__bridge CFStringRef).

I am going to test if it actually works now.

**Comment by renatorozas** [comment]
Confirmed, it works.

But I still don't know what a Bridge Cast. I guess that's homework for me.

I'll update this library for ARC projects and then push it somewhere.

Contact me if you want:

Twitter: @renatomoya

**Comment by bjornsallarp** [comment]
Hi,

Super late response. I just tested this and I have no problems. With "-fno-objc-arc" compiler flag added to the files it compiles and runs just fine. You mention that you removed autorelease etc, you should not do that with "-fno-objc-arc".

Tested using XCode 4.3.3.

**Comment by renatorozas** [comment]
Thanks Bjorn,

I believe the issue was about a C Bridge. As you say, with that flag, ARC compatibility is resolved.

Regards.
